test/realtime_server_goal_handle_tests.cpp

```diff
@@ -0,0 +1,280 @@
+ *       contributors may be used to endorse or promote products derived from
+ *       this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <gtest/gtest.h>
+#include <realtime_tools/realtime_server_goal_handle.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/TwoIntsAction.h>
+#include <chrono>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+using realtime_tools::RealtimeServerGoalHandle;
+
+const size_t ATTEMPTS = 10;
+const std::chrono::milliseconds DELAY(250);
+
+struct ActionCallback
+{
+  using GoalHandle = actionlib::ActionServer<actionlib::TwoIntsAction>::GoalHandle;
+  bool have_handle_ = false;
+  GoalHandle handle_;
+  std::mutex mtx_;
+
+  void goal_callback(GoalHandle handle)
+  {
+    std::unique_lock<std::mutex> lock(mtx_);
+    handle_ = handle;
+    handle_.setAccepted();
+    have_handle_ = true;
+  }
+
+  void cancel_callback(GoalHandle handle)
+  {
+  }
+
+  bool wait_for_handle()
+  {
+    for (size_t i = 0; i < ATTEMPTS; ++i) {
+      ros::spinOnce();
+      std::this_thread::sleep_for(DELAY);
+      std::unique_lock<std::mutex> lock(mtx_);
+      if (have_handle_) {
+        break;
+      }
+    }
+    std::unique_lock<std::mutex> lock(mtx_);
+    return have_handle_;
+  }
+};
+
+struct FeedbackCallback
+{
+  bool have_feedback_ = false;
+  std::mutex mtx_;
+
+  using FeedbackConstPtr = actionlib::ActionServer<actionlib::TwoIntsAction>::FeedbackConstPtr;
+  void feedback_callback(const FeedbackConstPtr &)
+  {
+    std::unique_lock<std::mutex> lock(mtx_);
+    have_feedback_ = true;
+  }
+
+  bool wait_for_feedback()
+  {
+    for (size_t i = 0; i < ATTEMPTS; ++i) {
+      ros::spinOnce();
+      std::this_thread::sleep_for(DELAY);
+      std::unique_lock<std::mutex> lock(mtx_);
+      if (have_feedback_) {
+        break;
+      }
+    }
+    std::unique_lock<std::mutex> lock(mtx_);
+    return have_feedback_;
+  }
+};
+
+std::shared_ptr<actionlib::SimpleActionClient<actionlib::TwoIntsAction>>
+send_goal(
+  const std::string & server_name,
+  FeedbackCallback * cb = nullptr)
+{
+  std::shared_ptr<actionlib::SimpleActionClient<actionlib::TwoIntsAction>> ac;
```

Comment:
Let's get [crazy](https://www.youtube.com/watch?v=u9sJ4NAugxg)!
```suggestion
  auto ac = std::make_shared<TwoIntsActionClient>(server_name, true);
```

and with this the next line can be dropped. (Looks fine to me but if it doesn't compile, you get a :cookie:  from Karsten)

Reply:
It compiled, no :cookie: for me :(

Use `std::make_shared` for server and client in 717f5473a782a9d04ef55550dc9afc331536ffa9

Reply:
Thanks!

Reply:
:+1: